### PR TITLE
feat(llm): add prompt caching configuration

### LINKED ISSUE #700: unrecognizedProperty cache_control
### OpenCodeReview Version

v1.8.5 (582953937) linux/amd64

### Operating System

Linux (x86_64)

### Installation Method

npm (global)

### LLM Provider

Anthropic (Claude)

### Bug Description

```
[ocr] Subtask error for pom.xml: LLM completion error: POST "<redacted>": 400 Bad Request {"errorCode":"INVALID_ARGUMENT","errorName":"LanguageModelService:InvalidRequest","errorInstanceId":"b5d73937-d0f4-4dbc-b5ce-1eb5bb3bd40c","parameters":{"unsafeParams":"{unrecognizedProperty=cache_control}","message":"Request contained an unrecognized field"}}
```
Is it possible to remove this cache_control via configuration?

BTW: `ocr llm test` was successful
```
Source: OCR environment
URL:    <redacted>
Model:  <redacted>
I am Open-Code-Review, a command-line code review assistant developed by Alibaba, designed to help analyze and review code directly in your terminal environment.
✓ Connection test successful
```

### Steps to Reproduce

-

### Expected Behavior

-

### Logs / Error Output

```shell

```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Linxiushen**
## Summary

- add a global `llm.prompt_caching` setting while preserving the current enabled-by-default behavior
- apply the setting to Anthropic system blocks and tool definitions across provider, legacy, and interactive configuration flows
- add CLI, resolver, request serialization, persistence, and TUI coverage, plus configuration documentation

## Testing

- `go test ./... -count=1 -skip 'TestResolveBackgroundFilePath|TestSaveConfig|TestResolveRuleEntries_SymlinkSafety|TestFinalizeSurfacesWriterCreationErrorWithoutStdout|TestHandleRepos_PermissionDenied|TestDiscoverRepos_SkipsUnreadableSubdir|TestListSessions_SkipsUnreadableFiles'`
- `go vet ./...`
- `go build -o .tmp/ocr.exe ./cmd/opencodereview`

Closes #700

### CODE REVIEW

File: cmd/opencodereview/provider_cmd.go [hunk 1]
```diff
@@ -113,6 +113,9 @@ func applyManualConfig(configPath string, cfg *Config, result providerTUIResult)
 		return fmt.Errorf("invalid auth_header: %w", err)
 	}
 	cfg.Llm.AuthHeader = authHeader
+	if result.promptCaching != nil {
+		cfg.Llm.PromptCaching = result.promptCaching
+	}
```
[the remark this answers is not in the document]
Comment (Linxiushen) [reply cmd/opencodereview/provider_cmd.go]: Addressed in c07b2a7. The manual, custom, and official apply paths now clear stale prompt-caching state when the effective protocol is not Anthropic. Regression coverage includes all three switch paths and preserves the active Anthropic setting when an inactive OpenAI provider is edited.

File: internal/llm/resolver.go [hunk 2]
```diff
@@ -420,16 +424,17 @@ func tryProviderConfig(cfg configFile, modelOverride string) (ResolvedEndpoint,
...
+		Model:         model,
+		Provider:      cfg.Provider,
+		Protocol:      protocol,
+		AuthHeader:    authHeader,
+		Source:        "provider:" + cfg.Provider,
+		ExtraBody:     extraBody,
+		ExtraHeaders:  extraHeaders,
+		PromptCaching: cfg.Llm.PromptCaching,
+		Timeout:       timeout,
 	}, true, nil
```
[the remark this answers is not in the document]
Comment (Linxiushen) [reply internal/llm/resolver.go:439]: Confirmed as global by design: the toggle follows whichever Anthropic provider is active rather than being duplicated across provider entries. c07b2a7 documents that contract in both the persisted config and resolver, while keeping provider-specific transport fields on each entry.

**Comment by cometkim** [comment]
I think this may duplicate https://github.com/alibaba/open-code-review/pull/332, which adds a generalized option for prompt caching, for multiple providers.

**Comment by lizhengfeng101** [comment]
Thanks a lot for this contribution, @Linxiushen, and for the thorough tests and docs that came with it. 🙏

After reviewing the open work around #700, we're going to move forward with #1038 instead, and I'd like to close this PR in its favor. The reasoning:

- **Granularity.** #700 is triggered by a *specific* Anthropic-compatible gateway that rejects `cache_control`. #1038 exposes the toggle at the provider level (`providers.<name>.prompt_cache` / `custom_providers.<name>.prompt_cache`) in addition to the global `llm` scope, so users can disable caching only for the misbehaving endpoint while keeping the cache discount on their other providers. This PR only offers the global switch.
- **Test coverage tailored to the bug.** #1038 asserts directly on the marshalled request body that `cache_control` is dropped (the exact wire-format issue in #700), plus disk round-trip and pointer-aliasing edge cases.
- **Status.** #1038 has already been reviewed and approved.

Really appreciate the effort here, and sorry for the overlap with a parallel PR. Closing in favor of #1038.
